datafusion/physical-expr/src/expressions/case.rs

```diff
@@ -195,8 +195,12 @@ impl CaseExpr {
                 _ => when_value,
             };
             let when_value = when_value.into_array(batch.num_rows());
-            let when_value = as_boolean_array(&when_value)
-                .expect("WHEN expression did not return a BooleanArray");
+            let when_value = as_boolean_array(&when_value).map_err(|e| {
+                DataFusionError::Context(
```

Comment:
🤔

Reply:
Yeah figured better to wrap in Context to hopefully provide more informative error message